tools/get_avg_prediction_time.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import json
+import argparse
+
+import numpy as np
+
+parser = argparse.ArgumentParser(description='Extract the average prediction\
+                                    time with the specified trace directory')
+parser.add_argument('trace_dir', type=str, help='Specify the trace directory')
+args = parser.parse_args()
+
+trace_dir = args.trace_dir
+trace_path = trace_dir + "/*.ctr"
+
+
+def parse_tracefile(filename):
+
+    with open(filename, 'r') as f:
+        raw = json.load(f)
+
+    if 'traceEvents' not in raw:
+        return None
+
+    traceEvents = raw['traceEvents']
+
+    timestamps = (
+        (
+            event['ts'],
+            event['ts'] + event['dur']
+        )
+        for event in traceEvents
+        if 'ts' in event and 'dur' in event
+    )
+    timestamps = sorted(timestamps, key=lambda x: x[1])
+
+    min_ts = timestamps[0]
+    max_ts = timestamps[-1]
+    return max_ts[1] - min_ts[0]
+
+
+durations = []
+
+for filename in glob.glob(trace_path):
+    # Exclude init0.ctr and prediction0.ctr as they don't reflect the real prediction time.
```

Comment:
ideally take a list of files as input instead since the names `init0.ctr` and `prediction0.ctr` are arbitrary

Reply:
Agreed, I will change this in a separate pr. Thank you.